centreon/www/front_src/src/Resources/Details/tabs/Graph/index.tsx

```diff
@@ -1,95 +1,56 @@
 import HostGraph from './HostGraph';
-import { GraphTimeParameters } from './models';
-
-const useStyles = makeStyles()((theme: Theme) => ({
-  container: {
-    display: 'grid',
-    gridRowGap: theme.spacing(2),
-    gridTemplateRows: 'auto 1fr'
-  },
-  exportToPngButton: {
-    display: 'flex',
-    justifyContent: 'space-between',
-    margin: theme.spacing(0, 1, 1, 2)
-  },
-  graph: {
-    height: '100%',
-    margin: 'auto',
-    width: '100%'
-  },
-  graphContainer: {
-    display: 'grid',
-    gridTemplateRows: '1fr',
-    padding: theme.spacing(2, 1, 1)
-  }
-}));
+import { updatedGraphIntervalAtom } from './atoms';
+import { useChartGraphStyles } from './chartGraph.styles';
 
 const GraphTabContent = ({ details }: TabProps): JSX.Element => {
-  const { classes } = useStyles();
+  const { classes } = useChartGraphStyles();
 
-  const [graphTimeParameters, setGraphTimeParameters] =
-    useState<GraphTimeParameters>();
+  const [updatedGraphInterval, setUpdatedGraphInterval] = useAtom(
+    updatedGraphIntervalAtom
+  );
+
+  const [graphTimeParameters, setGraphTimeParameters] = useState<Parameters>();
 
   const type = details?.type as ResourceType;
   const equalsService = equals(ResourceType.service);
   const equalsMetaService = equals(ResourceType.metaservice);
   const equalsAnomalyDetection = equals(ResourceType.anomalyDetection);
-  const updatedGraphInterval = useAtomValue(updatedGraphIntervalAtom);
 
   const isService =
     equalsService(type) ||
     equalsMetaService(type) ||
     equalsAnomalyDetection(type);
 
-  const getTimePeriodsParameters = (data: GraphTimeParameters): void => {
+  const getTimePeriodsParameters = (data: Parameters): void => {
     setGraphTimeParameters(data);
   };
 
-  const newGraphInterval = updatedGraphInterval
-    ? { end: updatedGraphInterval.end, start: updatedGraphInterval.start }
-    : undefined;
-
   return (
-    <div className={classes.container}>
+    <div className={classes.graphTabContainer}>
       {isService ? (
         <>
           <TimePeriods
-            adjustTimePeriodData={newGraphInterval}
+            adjustTimePeriodData={updatedGraphInterval}
             getParameters={getTimePeriodsParameters}
             renderExternalComponent={<GraphOptions />}
           />
-          {graphTimeParameters && (
-            <ExportablePerformanceGraphWithTimeline
-              interactWithGraph
-              graphHeight={280}
-              graphTimeParameters={graphTimeParameters}
-              renderAdditionalLines={(props): JSX.Element => (
-                <FederatedComponent
-                  {...props}
-                  path="/anomaly-detection/thresholdLines"
-                />
-              )}
-              resource={details}
-            />
-          )}
+
+          <ChartGraph
+            resource={details}
+            graphTimeParameters={graphTimeParameters}
+            updatedGraphInterval={setUpdatedGraphInterval}
+          />
```

Comment:
- the conditional rendering based on the state isService is clear , showing if a resource of type another than service we display TimePeriods & chart graph otherwise we display HostGraph (like the name explain graph of Hosts)

- setGraphTimeParameters its well used , like the name suggest getting the parametrs of TimePeriod component